pkg/cmd/issue/shared/lookup.go

```diff
@@ -13,10 +13,89 @@ import (
 	"github.com/cli/cli/v2/api"
 	fd "github.com/cli/cli/v2/internal/featuredetection"
 	"github.com/cli/cli/v2/internal/ghrepo"
+	o "github.com/cli/cli/v2/pkg/option"
 	"github.com/cli/cli/v2/pkg/set"
 	"golang.org/x/sync/errgroup"
 )
 
+func ParseIssuesFromArgs(args []string) ([]int, o.Option[ghrepo.Interface], error) {
+	var repo o.Option[ghrepo.Interface]
+	issueNumbers := make([]int, len(args))
+
+	for i, arg := range args {
+		// For each argument, parse the issue number and an optional repo
+		issueNumber, issueRepo, err := parseIssueFromArg(arg)
+		if err != nil {
+			return nil, o.None[ghrepo.Interface](), err
+		}
+
+		// if this is our first issue repo found, then we need to set it
+		if repo.IsNone() {
+			repo = issueRepo
+		}
+
+		// if there is an issue repo returned, then we need to check if it is the same as the previous one
+		if issueRepo.IsSome() && repo.IsSome() {
+			// Unwraps are safe because we've checked for presence above
+			if !ghrepo.IsSame(repo.Unwrap(), issueRepo.Unwrap()) {
+				return nil, o.None[ghrepo.Interface](), fmt.Errorf(
+					"multiple issues must be in same repo: found %q, expected %q",
+					ghrepo.FullName(issueRepo.Unwrap()),
+					ghrepo.FullName(repo.Unwrap()),
+				)
+			}
+		}
+
+		// add the issue number to the list
+		issueNumbers[i] = issueNumber
+	}
+
+	return issueNumbers, repo, nil
+}
+
+func parseIssueFromArg(arg string) (int, o.Option[ghrepo.Interface], error) {
+	issueLocator := tryParseIssueFromURL(arg)
+	if issueLocator, present := issueLocator.Value(); present {
+		return issueLocator.issueNumber, o.Some(issueLocator.repo), nil
+	}
+
+	issueNumber, err := strconv.Atoi(strings.TrimPrefix(arg, "#"))
+	if err != nil {
+		return 0, o.None[ghrepo.Interface](), fmt.Errorf("invalid issue format: %q", arg)
+	}
+
+	return issueNumber, o.None[ghrepo.Interface](), nil
+}
+
+type issueLocator struct {
+	issueNumber int
+	repo        ghrepo.Interface
+}
+
+// tryParseIssueFromURL tries to parse an issue number and repo from a URL.
+func tryParseIssueFromURL(maybeURL string) o.Option[issueLocator] {
+	u, err := url.Parse(maybeURL)
+	if err != nil {
+		return o.None[issueLocator]()
+	}
+
+	if u.Scheme != "https" && u.Scheme != "http" {
```

Comment:
Bare numbers parse as valid URL, so I guess this is checking against that. Probably needs a comment.

Reply:
Yes, please.